feat(clp-s): Automatically infer the file types of inputs to compression. (#1312)
y-scope/clp: docs/src/user-docs/core-clp-s.md

* `archives-dir` is the directory that archives should be written to.
* `input-path` is a filesystem path or URL to either:

>   * a new-line-delimited JSON (ndjson) log file;
>   * a KV-IR file;
>   * either of these file-types compressed by Zstd; or

Reviewer: ```suggestion
  * a new-line-delimited JSON (ndjson) log file (may be Zstd-compressed);
  * a KV-IR file (may be Zstd-compressed);
```
